### PR TITLE
Fix sliding when resizing with minimum limits reached

### LINKED ISSUE #3326: Unskip resizing test after bugfix
Since #3199 got resolved with #3259, we need to unskip the relevant e2e test for it:

https://github.com/ampproject/amp-wp/blob/b44ac6ac687008e4d1c5dd2ef78b2a5d3d9ecd23/tests/e2e/specs/stories-editor/resizing.js#L275-L299

In my quick test it actually didn't quite work out of the box, so perhaps the issue is not yet fully resolved.

### LINKED ISSUE #3397: Resizing Text block: Trying to resize smaller than possible moves the position
Steps to reproduce:
- Create a Text block.
- Turn off amp-fit-text.
- Resize it from the left handle to as narrow as possible.
- See the results.
resizing-2

Expected: The block should stay in place when resizing and reaching limits.

Note: Same issue for non-rotated Text block, too.

Todo:
- [x] Fix the issue
- [x] Add tests for resizing for when amp fit text is turned off.

### PR DESCRIPTION
**Pull request by miina**
## Summary

Fixes #3397
Fixes #3326 

- Unskips the resizing tests.
- Prevents position change when minimum limits have been reached when reducing the width/height.
- Prevents `scrollHeight` calculations to limit resizing if the Text block is empty anyway.

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/issues/3397).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md).

### CODE REVIEW

**Comment by miina** [comment]
OK, found the reason for tests being unreliable: there was an `await` missing 🤦‍♀ 

However, this brought out some actual additional bug when fit-text is turned off, so looking into that now.

File: tests/e2e/specs/stories-editor/resizing.js [hunk 1]
```diff
@@ -269,34 +269,78 @@ describe( 'Resizing', () => {
...
+			// Get the initial position.
+			const { positionTop: positionTopBefore } = await getSelectedBlockPosition();
+			// Try resizing again.
+			await dragAndResize( resizableHandleTop, { x: 0, y: 250 } );
+			const { positionTop } = await getSelectedBlockPosition();
+			expect( positionTop ).toStrictEqual( positionTopBefore );
+
+			// Rotate the block and try again.
+			await rotateSelectedBlock();
+			await dragAndResize( resizableHandleTop, { x: 300, y: 0 } );
```
Comment (barklund) [inline tests/e2e/specs/stories-editor/resizing.js]: Here you switch to dragging in `x` direction in stead of `y` direction after rotation. But in the above test, you drag in `x` both before and after rotation. I don't know which is the right one, but it seems to be this one, perhaps?

I guess that depends on how the item is rotated of course 😝.

**Comment by barklund** [comment]
If you resize "quickly", then the item jumps when you reach minimum position, as it jumps to the last "legal" top position and minimum height from there in stead of minimum height based off the "static" bottom position.

This is particularly obvious with a rotated block, but happens for non-rotated ones as well.

I tried to see, if I could fix this, but I just can't wrap my head around it. Maybe for non-rotated blocks, but definitely couldn't figure it out for the rotated ones.

Screen Recording 2019-10-03 at 17 07 27

**Comment by miina** [comment]
Thanks for testing! I think I remember seeing this "jumpiness" on quick resizing from before as well, it's not easy to test on `develop` though since currently the block slides away instead (without this PR).

Will try to look into this jumpiness as well in this PR, if it starts taking too much time then will create a separate issue instead.

File: tests/e2e/specs/stories-editor/resizing.js (hunk 1 above)
Comment (miina) [reply tests/e2e/specs/stories-editor/resizing.js]: Hah! Good catch, thank you. The rotation is not enough for the initial direction not to work -- will leave a comment with degrees.

**Comment by spacedmonkey** [comment]
I am going to wait to code review until the  "jumpiness"  is fixed.

**Comment by miina** [comment]
Added a new issue for the jumpiness. It's unrelated to the original issue anyway and possibly lower priority, too.
https://github.com/ampproject/amp-wp/issues/3444

**Comment by swissspidy** [comment]
Still seeing the issue: https://cloudup.com/cRZMrlUPdy8

**Comment by miina** [comment]
> Still seeing the issue: https://cloudup.com/cRZMrlUPdy8

Hmm, that seems to be a different issue -- the original issue for the position continuing to change even though the minimum limits have been reached and it's not possible to reduce anymore (in case of amp-fit-text turned off).

I'll debug this new finding as well to see if it seems to be an easy fix. For testing/QA purposes might be good to track that as a different issue though.

Looks like that's for when "amp-fit-text" is turned on, right?

**Comment by swissspidy** [comment]
> Looks like that's for when "amp-fit-text" is turned on, right?

Correct. A separate issue is fine for me. Thanks for having a look!

**Comment by miina** [comment]
OK, looks like the tests have passed a few times in a row now.

**Pull request merged**